src/slang/utils/rttc.ts

```diff
@@ -51,15 +51,12 @@ export const checkUnaryExpression = (
   value: Value
 ) => {
   const node = context.runtime.nodes[0]
-  switch (operator) {
-    case '+':
-      return isNumber(value) ? undefined : new TypeError(node, '', 'number', typeOf(value))
-    case '-':
-      return isNumber(value) ? undefined : new TypeError(node, '', 'number', typeOf(value))
-    case '!':
-      return isBool(value) ? undefined : new TypeError(node, '', 'boolean', typeOf(value))
-    default:
-      return
+  if ((operator === '+' || operator === '-') && !isNumber(value)) {
```

Comment:
I feel like the switch case used previously is clearer, although simplifying the logic would help with more operators

Reply:
So the thing is that with the switch case, in order to get 100% test coverage there would need to be a call to `checkUnaryExpression(context, ''. value)`. Otherwise, the default case would not be covered by the tests. 

So assuming we want 100% test coverage, it was either that, or to change the switch case to the if/else here. Since there isn't much sense in checking for an empty operator, I decided to go with the latter instead.

Reply:
Can the default case be used to test for an unspecified unary operator (e.g `delete` or `typeof`)? If so, it can be meaningfully used

Reply:
That sounds like a good idea, tbh. 

Unspecified operators are currently checked for in the linting step at at [`noUnspecifiedOperators`](https://github.com/source-academy/cadet-frontend/blob/1559e07a196e3260fafa35ac308cf589861f2c10/src/slang/rules/noUnspecifiedOperator.ts). Checking the operators again during runtime seems redundant at first, but I feel like it's extra safety without any real performance hits.

Reply:
@evansb Can we check, what's the best practice here?